tests/test_template.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :mod:`pennylane.template` module.
+"""
+# pylint: disable=protected-access,cell-var-from-loop
+import unittest
+import inspect
+import logging as log
+import scipy as sp
+from scipy.linalg import qr
+
+from pennylane import numpy as np
+
+from defaults import pennylane as qml, BaseTest
+
+from unittest.mock import patch
+from pennylane.ops import Kerr
+
+log.getLogger('defaults')
+
+class TestInterferometer(BaseTest):
+    """Tests for the Interferometer from the pennylane.template module."""
+
+    num_subsystems = 4
+
+    def randnc(self, *arg):
+        """Normally distributed array of random complex numbers."""
+        return np.random.randn(*arg) + 1j*np.random.randn(*arg)
+
+    def setUp(self):
+        super().setUp()
+        np.random.seed(35)
+        self.u1 = self.random_interferometer(self.num_subsystems)
+        self.u2 = self.random_interferometer(self.num_subsystems-1)
+
+    def test_interferometer(self):
+        dev = qml.device('default.gaussian', wires=self.num_subsystems)
+        np.random.seed(8)
+        squeezings = np.random.rand(self.num_subsystems, 2)
+        num_params = int(self.num_subsystems*(self.num_subsystems-1)/2)
+        theta = np.random.uniform(0, 2*np.pi, num_params)
+        phi = np.random.uniform(0, 2*np.pi, num_params)
+
+        @qml.qnode(dev)
+        def circuit(theta, phi):
+            for wire in range(self.num_subsystems):
+                qml.Squeezing(squeezings[wire][0], squeezings[wire][1], wires=wire)
+
+            qml.template.Interferometer(theta=theta, phi=phi, wires=range(self.num_subsystems))
+            return tuple(qml.expval.MeanPhoton(wires=wires) for wires in range(self.num_subsystems))
+
+        self.assertAllAlmostEqual(circuit(theta, phi), [0.96852694, 0.23878521, 0.82310606, 0.16547786], delta=self.tol)
+
+        self.assertAllAlmostEqual(qml.jacobian(circuit, 0)(theta, phi),
```

Comment:
same question

Reply:
Also this is just an execution test, i.e., does the code run without problems, and not a proper unit test.